utils/multi-step-transform/index.js

```diff
@@ -0,0 +1,52 @@
+const { default: PQueue } = require('p-queue');
+const OOOTransform = require('yeoman-environment/lib/util/out-of-order-transform');
+const { isFilePending } = require('mem-fs-editor/lib/state');
+
+const { TemplateFileFs } = require('./template-file-fs');
+
+module.exports.MultiStepTransform = class MultiStepTransform extends OOOTransform {
+  constructor(options = {}) {
+    super(options);
+
+    this.twoStepTemplateQueue = new PQueue({ concurrency: 1, autoStart: false });
```

Comment:
Can't we just use this.queue instead of creating a brand new one?

Reply:
Unfortunately no, we need to empty the `this.queue` before emptying ours.
So every fragment files is registered.